src/opaque.c

```diff
@@ -1304,10 +1309,10 @@ int opaque_CreateCredentialResponse(const uint8_t _pub[OPAQUE_USER_SESSION_PUBLI
   dump((uint8_t*)preamble, sizeof preamble, "auth preamble");
 #endif
   if(NULL!=authU) {
-    crypto_auth_hmacsha512(authU,                               // out
-                           (uint8_t*)preamble,                  // in
-                           crypto_hash_sha512_BYTES,            // len(in)
-                           keys.km3);                           // key
+    opaque_hmacsha512(keys.km3,                       // key
+                     (uint8_t*)preamble,              // in
+                     crypto_hash_sha512_BYTES,        // len(in)
+                     authU);                          // out
```

Comment:
woah. really? if this is true. that's very embarrassing.

Reply:
oh, i see it's also a different function. ok. that explains the different order of params. phew.

Reply:
Yeah, I had to use a different function here, don't worry :D